packages/aws-client/src/DynamoDb.ts

```diff
@@ -147,49 +148,69 @@ export const scan = improveStackTrace(
  * See https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Scan.html#Scan.ParallelScan.
  * See [DocumentClient.scan()](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/DynamoDB/DocumentClient.html#scan-property).
  *
- * @param {number} totalSegments
+ * @param {Object} params
+ * @param {number} params.totalSegments
  *   Total number of segments to divide table into for parallel scanning
- * @param {DocumentClient.ScanInput} scanParams
+ * @param {DocumentClient.ScanInput} params.scanParams
  *   Params for the DynamoDB client scan operation
  *   See https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Scan.html
- * @param {function} processItemsFunc - Function used to process returned items by scan
+ * @param {function} params.processItemsFunc - Function used to process returned items by scan
+ * @param {DocumentClient} [params.dynamoDbClient] - Instance of Dynamo DB document client
+ * @param {pRetry.Options} [params.retryOptions] - Retry options for scan operations
  * @returns {Promise}
  */
 export const parallelScan = async (
-  totalSegments: number,
-  scanParams: DocumentClient.ScanInput,
-  processItemsFunc: (items: DocumentClient.ItemList) => Promise<void>
-) => pMap(
-  range(totalSegments),
-  async (_, segmentIndex) => {
-    let exclusiveStartKey: DocumentClient.Key | undefined;
-
-    const segmentScanParams: DocumentClient.ScanInput = {
-      ...scanParams,
-      TotalSegments: totalSegments,
-      Segment: segmentIndex,
-    };
-
-    /* eslint-disable no-await-in-loop */
-    do {
-      const {
-        Items = [],
-        LastEvaluatedKey,
-      } = await dynamodbDocClient().scan(segmentScanParams).promise();
-
-      exclusiveStartKey = LastEvaluatedKey;
-      segmentScanParams.ExclusiveStartKey = exclusiveStartKey;
-
-      await processItemsFunc(Items);
-    } while (exclusiveStartKey);
-    /* eslint-enable no-await-in-loop */
-
-    return Promise.resolve();
-  },
-  {
-    stopOnError: false,
+  params: {
+    totalSegments: number,
+    scanParams: DocumentClient.ScanInput,
+    processItemsFunc: (items: DocumentClient.ItemList) => Promise<void>,
+    dynamoDbClient?: DocumentClient,
+    retryOptions?: pRetry.Options,
   }
-);
+) => {
+  const {
+    totalSegments,
+    scanParams,
+    processItemsFunc,
+    dynamoDbClient = dynamodbDocClient(),
+    retryOptions,
+  } = params;
+
+  return pMap(
+    range(totalSegments),
+    async (_, segmentIndex) => {
+      let exclusiveStartKey: DocumentClient.Key | undefined;
+
+      const segmentScanParams: DocumentClient.ScanInput = {
+        ...scanParams,
+        TotalSegments: totalSegments,
+        Segment: segmentIndex,
+      };
+
+      /* eslint-disable no-await-in-loop */
+      do {
+        const {
+          Items = [],
+          LastEvaluatedKey,
+        } = await pRetry(
+          () => dynamoDbClient.scan(segmentScanParams).promise(),
```

Comment:
Github made this diff a bit onerous, but this is just adding a passed in doc client and retry options and setting pmap.    👍🏻